sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -592,3 +593,18 @@ case class QualifyLocationWithWarehouse(catalog: SessionCatalog) extends Rule[Lo
       c.copy(tableDesc = newTable)
   }
 }
+
+object CollationCheck extends (LogicalPlan => Unit) {
+  def apply(plan: LogicalPlan): Unit = {
+    plan.foreach {
```

Comment:
we can call `transformAllExpressionsWithSubqueries`

Reply:
I tried doing this, but I get error that transformAllExpressionsWithSubqueries cannot be called in analyzer.

Reply:
I think I added it now, could you check if this is what you were expecting?

Reply:
where do you match `SubqueryExpression`?

Reply:
This is not resolved yet. `SubqueryExpression` is an expression that contains query plans, we need to recurse into the query plan from it.

Reply:
oh nvm, the framework calls check rules for subquery expressions already.

Reply:
Aha, I thought we avoided `SubqueryExpression` needs with the addition of operator.expressions.foreach. Will look into it now.